src/prototype/main.py

```diff
@@ -0,0 +1,85 @@
+import sys
+import re
+import os
+from PyQt5.QtWidgets import QApplication, QMainWindow
+from PyQt5.QtCore import pyqtSlot
+from gui.mainwidow_ui import Ui_MainWindow
+from OdasStream.odas_stream import OdasStream
+from FileHelper.file_helper import FileHelper
+from ArgsParser.args_parser import ArgsParser
+
+
+class MainWindow(QMainWindow, Ui_MainWindow):
+    def __init__(self, odasStream, parent=None):
+        super(MainWindow, self).__init__(parent)
+        self.setupUi(self)
+        self.odasStream = odasStream
+
+        # signal slots
+        self.odasStream.signalOdasData.connect(self.odasDataReveived)
+        self.btnStartOdas.clicked.connect(self.btnStartOdasClicked)
+        self.btnStopOdas.clicked.connect(self.btnStopOdasClicked)
+
+
+    def closeEvent(self, event):
+        self.stopOdas()
+        event.accept()
+
+
+    def startOdas(self):
+        if self.odasStream and not self.odasStream.isRunning:
+            self.odasStream.start()
+
+
+    def stopOdas(self):
+        if self.odasStream and self.odasStream.odasProcess:
+            if self.odasStream.isRunning:
```

Comment:
I would put the second if with the first one. If you want to put it in two lines you can use:
if self.odasStream and self.odasStream.odasProcess \\
  and self.odasStream.isRunning: